src/lib/verifier/DAVerifier.sol

```diff
@@ -235,11 +240,21 @@ library DAVerifier {
     /// @param _begin The beginning of the range (inclusive).
     /// @param _end The ending of the range (exclusive).
     /// @return _ the sliced data.
-    function slice(bytes[] memory _data, uint256 _begin, uint256 _end) internal pure returns (bytes[] memory) {
+    function slice(bytes[] memory _data, uint256 _begin, uint256 _end)
+        internal
+        pure
+        returns (bytes[] memory, ErrorCodes)
+    {
+        if (_begin > _end) {
+            return (_data, ErrorCodes.InvalidRange);
+        }
+        if (_begin > _data.length || _end > _data.length) {
+            return (_data, ErrorCodes.OutOfBoundsRange);
+        }
         bytes[] memory out = new bytes[](_end-_begin);
         for (uint256 i = _begin; i < _end; i++) {
-            out[i] = _data[i];
+            out[i - _begin] = _data[i];
```

Comment:
Can you add a test that hits this codepath and would have caught the bug?

Reply:
[a042ac6](https://github.com/celestiaorg/blobstream-contracts/pull/254/commits/a042ac6b1124fa9880c779b8ff655a02164e83f5)